rfc/2020/scale-testing/kohsheen1234/rfc.md

```diff
@@ -0,0 +1,53 @@
+- Contribution Name: Scale Testing
+- Implementation Owner:  ['kohsheen1234'](https://github.com/kohsheen1234)
+- Start Date: 29-02-2020
+- Target Date: 
+- RFC PR:
+- Linkerd Issue: [linkerd/linkerd2#3895](https://github.com/linkerd/linkerd2/issues/3895)
+- Reviewers: 
+
+[summary]: #summary
+
+# Summary
+This conrtibution aims to completely automate the scale tests, so that scale-testing can be performed very easily. Automating scale tests is done to ensure the performance of Linkerd is maintained.
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+- Identify performance regression/potential errors with minimal manual intervention.
+- Latency in the network communication can be a massive issue. When new features added to Linkerd, it might detoirate the performance of Linkerd. So, it is important to understand if any new feature which has been added recently has a negative impact on Linkerd's performance. Manually checking every-time if there has been a regression in the performance is not a convinient solution. This process can be automated, so that it is very simple to perform scale testing.
+- Once this scale testing has been automated, Linkerd will be able to :
+  - Automatically add a sample workload to the cluster.
```

Comment:
In phase 2, consider whether your test framework will automate the installation of Linkerd and other infrastructure resources like Kubernetes cluster, tainted nodes to run Prometheus etc.